src/briefcase/platforms/linux/system.py

```diff
@@ -892,6 +893,10 @@ def run_app(
                 )
 
 
+class LinuxSystemDevCommand(LinuxSystemMixin, DevCommand):
```

Comment:
In testing, I found an interesting failure mode of this specific inheritance heirarchy: If you run `briefcase dev linux` (or `briefcase dev linux system`) on a macOS machine, the error message says:
```
Linux system projects can only be built on Linux, or on macOS using Docker.
```

This is true... for full packages. However, in the context of dev mode, it's a little misleading. We aren't going to be able to add Docker support for Linux dev mode without a *lot* more work (which is *way* outside scope for what you're trying to do here). 

I suspect the base command here needs to be `LinuxMixin`, so that the Docker behaviors aren't picked up. This may also require defining `supported_host_os`, and a few other keys (see the definition of LinuxSystemPassiveMixin for details).